tracing-subscriber/src/fmt/format/json.rs

```diff
@@ -236,6 +236,28 @@ where
                 serializer.serialize_entry("target", meta.target())?;
             }
 
+            if self.display_thread_name {
+                match std::thread::current().name() {
+                    Some(name) => {
+                        serializer.serialize_entry("threadName", name)?;
+                    }
+                    None => {
+                        // fall-back to thread id when name is absent and id's are not enabled
+                        if !self.display_thread_id {
+                            serializer.serialize_entry(
+                                "threadId",
+                                &format!("{:?}", std::thread::current().id()),
```

Comment:
nit: it might be slightly more efficient to avoid getting the current thread twice in this case:
```suggestion
                let thread = std::thread::current();
                match thread.name() {
                    Some(name) => {
                        serializer.serialize_entry("threadName", name)?;
                    }
                    None => {
                        // fall-back to thread id when name is absent and id's are not enabled
                        if !self.display_thread_id {
                            serializer.serialize_entry(
                                "threadId",
                                &format!("{:?}", thread.id()),
```
but, this may not matter a whole lot...

Reply:
Hmm, This is interesting but only one path gets executed right? So, I don't see the value but ti def helps if we get thread outside the `if self.display_thread_name` as we can re-use it for `if self.display_thread_id` but that would mean even if both these flags are not enabled the user is incurring the cost of `thread.current()`

Maybe I'm overthinking here :p